src/main/java/com/epam/reportportal/cucumber/RunningContext.java

```diff
@@ -141,7 +157,8 @@ void setFeatureId(Maybe<String> featureId) {
     }
 
     static class ScenarioContext {
-        private static Map<String, String> outlineIterationsMap = new HashMap<String, String>();
+        private final static Object OUTLINE_SYNC = new Object();
```

Comment:
You are using `ConcurrentHashMap` it's already concurrent, you don't need this lock.